packages/reporter/src/main.tsx

```diff
@@ -89,7 +89,7 @@ class Reporter extends Component<SingleReporterProps> {
   // this hook will only trigger if we switch spec file at runtime
   // it never happens in normal e2e but can happen in component-testing mode
   componentDidUpdate (newProps: BaseReporterProps) {
-    this.props.runnablesStore.setRunningSpec(this.props.runnerStore.spec.relative)
+    this.props.runnablesStore.setRunningSpec(this.props.runnerStore.spec?.relative ?? null)
```

Comment:
Is it possible to update the setRunnerSpec method to accept undefined/null?

Reply:
It is, but this leads to needing to change a bunch of things in reporter that expect `null`. This change leads to the least amount of code changes.

This was not nullable before, since the type was overrode here: https://github.com/cypress-io/cypress/pull/20663/files#diff-8ec8396907f7c8abf3999a7ec920f7210d6730c10f9a30c9df87253b0b28c125L37 like this:

```ts
runnerStore: BaseStore & {spec: Cypress.Spec, specRunId: string}
```

I changed this to avoid having multiple types for the store spread over the app. If `spec` is really not allowed to be null here, I think it would make sense to pass it as a separate prop, rather than requiring an entire store with several type changes, so we don't end up in a situation where the `BaseStore` is augmented all over the place.

It's not entirely clear to me that we can actually guarantee `spec` is not null, and I didn't want to change too much here, so this seemed like the past of least resistance/lowest risk.